projects/admin-core/src/lib/application/state/application.selectors.ts

```diff
@@ -203,3 +208,37 @@ export const selectServiceLayerTreeForApplication = createSelector(
 export const selectStylingConfig = createSelector(selectDraftApplication, application => application?.styling);
 
 export const selectFilterGroups = createSelector(selectDraftApplication, application => application?.settings?.filterGroups || []);
+
+export const selectFilterableLayersForApplication = createSelector(
+  selectAppLayerNodesForSelectedApplication,
+  selectGeoServiceLayers,
+  selectGeoServices,
+  selectFeatureTypes,
+  (appLayersNodes, geoServiceLayers, geoServices, featureTypes): GeoServiceLayerInApplicationModel[] => {
+    const geoServiceLayerMap = ApplicationTreeHelper.getLayerMap(geoServiceLayers);
+    return appLayersNodes
+      .filter(layer => ApplicationModelHelper.isLayerTreeNode(layer))
+      .map(layerNode => ({
+        geoServiceLayer: geoServiceLayerMap.get(ApplicationTreeHelper.getLayerMapKey(layerNode.layerName, layerNode.serviceId)),
+        appLayerId: layerNode.id,
+      }))
+      .filter(geoServiceLayerInApplication => geoServiceLayerInApplication.geoServiceLayer !== undefined)
```

Comment:
Why not combine this filter with the last filter (and then filter first, then map)